File: VB/TabbedMdiManagerHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTab
Imports DevExpress.XtraTabbedMdi
Imports System.Drawing
Imports System.Collections

Namespace TabbedMDI
	Public Class TabbedMdiManagerHelper
		Private manager As XtraTabbedMdiManager
		Private clickTick As Integer = -1
		Private previousPage As IXtraTabPage

		Public Delegate Sub TabHeaderDoubleClickEventHandler(ByVal sender As Object, ByVal e As TabHeaderDoubleClickEventArgs)
		Public Event TabHeaderDoubleClick As TabHeaderDoubleClickEventHandler

		Public Sub New(ByVal manager As XtraTabbedMdiManager)
			Me.manager = manager
			AddHandler manager.MouseDown, AddressOf OnXtraTabbedMdiManagerMouseDown
		End Sub

		Private Sub OnXtraTabbedMdiManagerMouseDown(ByVal sender As Object, ByVal e As System.Windows.Forms.MouseEventArgs)
			Dim hi As DevExpress.XtraTab.ViewInfo.BaseTabHitInfo = manager.CalcHitInfo(New Point(e.X, e.Y))
			If hi.HitTest = DevExpress.XtraTab.ViewInfo.XtraTabHitTest.PageHeader Then
				If clickTick = -1 Then
					clickTick = System.Environment.TickCount
					previousPage = hi.Page
				Else
					If System.Environment.TickCount - clickTick < SystemInformation.DoubleClickTime AndAlso hi.Page Is previousPage Then
						Dim args As New TabHeaderDoubleClickEventArgs(hi.Page)
						RaiseTabHeaderDoubleClick(args)
					End If
					clickTick = -1
				End If
			End If
		End Sub

		Private Sub RaiseTabHeaderDoubleClick(ByVal e As TabHeaderDoubleClickEventArgs)
			RaiseEvent TabHeaderDoubleClick(manager, e)
		End Sub
	End Class
End Namespace

File: CS/TabbedMdiManagerHelper.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTab;
using DevExpress.XtraTabbedMdi;
using System.Drawing;
using System.Collections;

namespace TabbedMDI
{
    public class TabbedMdiManagerHelper
    {
        XtraTabbedMdiManager manager;
        int clickTick = -1;
        IXtraTabPage previousPage;

        public delegate void TabHeaderDoubleClickEventHandler(object sender, TabHeaderDoubleClickEventArgs e);
        public event TabHeaderDoubleClickEventHandler TabHeaderDoubleClick;

        public TabbedMdiManagerHelper(XtraTabbedMdiManager manager)
        {
            this.manager = manager;
            manager.MouseDown += OnXtraTabbedMdiManagerMouseDown;
        }

        private void OnXtraTabbedMdiManagerMouseDown(object sender, System.Windows.Forms.MouseEventArgs e)
        {
            DevExpress.XtraTab.ViewInfo.BaseTabHitInfo hi = manager.CalcHitInfo(new Point(e.X, e.Y));
            if (hi.HitTest == DevExpress.XtraTab.ViewInfo.XtraTabHitTest.PageHeader)
            {
                if (clickTick == -1)
                {
                    clickTick = System.Environment.TickCount;
                    previousPage = hi.Page;
                }
                else
                {
                    if (System.Environment.TickCount - clickTick < SystemInformation.DoubleClickTime && hi.Page == previousPage)
                    {
                        TabHeaderDoubleClickEventArgs args = new TabHeaderDoubleClickEventArgs(hi.Page);
                        RaiseTabHeaderDoubleClick(args);
                    }
                    clickTick = -1;
                }
            }
        }

        private void RaiseTabHeaderDoubleClick(TabHeaderDoubleClickEventArgs e)
        {
            if (TabHeaderDoubleClick != null)
                TabHeaderDoubleClick(manager, e);
        }
    }
}
